frontend/src/components/features/conversation-panel/conversation-panel.tsx

```diff
@@ -44,16 +43,10 @@ export function ConversationPanel({ onClose }: ConversationPanelProps) {
 
   const handleConfirmDelete = () => {
     if (selectedConversationId) {
-      deleteConversation(
-        { conversationId: selectedConversationId },
-        {
-          onSuccess: () => {
-            if (cid === selectedConversationId) {
-              endSession();
-            }
-          },
-        },
-      );
+      deleteConversation({ conversationId: selectedConversationId });
+      if (selectedConversationId === currentConversationId) {
+        navigate("/");
+      }
```

Comment:
```suggestion
      deleteConversation(
        { conversationId: selectedConversationId },
        {
          onSuccess: () => {
            if (selectedConversationId === currentConversationId) {
              navigate("/");
            }
          },
        },
      );
```

Only navigate if successful